tools/src/main/scala/scala/scalanative/build/LLVM.scala

```diff
@@ -331,4 +331,41 @@ private[scalanative] object LLVM {
     else str
   }
 
+  private def constructIdent(config: Config): String = {
+    val mt = config.compilerConfig.multithreadingSupport
+    val snVersion = scala.scalanative.nir.Versions.current
+
+    val ident1 = s"Scala Native ${snVersion}"
+    val ident2 = s"Multithread: ${mt},"
+    val ident3 = s"Mode: ${config.mode}, LTO: ${config.LTO}, GC: ${config.gc}"
+
+    s"${ident1} (${ident2} ${ident3})"
+  }
+
+  private[scalanative] def generateLLVMIdent(
+      config: Config
+  ): Seq[java.nio.file.Path] = {
+
+    /* Enable feature only where known to work. Add to list as experience grows
+     * FreeBSD uses elf format so it _should_ work, but it has not been
+     * exercised.
+     */
+    if (!config.targetsLinux) Seq.empty[Path]
```

Comment:
This part of code gen should be safe in all platforms and probably we can always include identity. 

When it comes to extracting identity using `readelf --string-dump` on Windows ChatGPT suggests: 
```
On Windows, the equivalent command to extract similar information from an executable file is objdump -s -j .comment binary-file.exe. Here's a brief explanation of the options used:

-s prints the contents of the specified sections.
-j specifies the section to print.
.comment is the name of the section that contains the compiler and linker version information.
Replace binary-file.exe with the name of your executable file. The output will contain the compiler and linker version information, similar to the output of readelf --string-dump .comment command on Linux. Note that the objdump command is available on most Unix-like systems as well, including Linux.
```
Have not tried it yet though

Reply:
Thank you for the encouragement, it sustains the effort.

>  but that's the opportunity for another day

Amen!   I am trying to focus on the JSR-166 work and not go haring off after this
bright shining ball.  It does sparkle though!

Interesting source of info on Windows.  I am familiar with the `bsdutils` package. For this PR,
I tried it on macOS and ran out of time before getting anything useful for this PR.
 
I will have to try `objdump` on Windows the next time I boot that system.  See if it exists and
if the compiler (Clang), handles the `LLVM.ident`.